migrations/migration.py

```diff
@@ -0,0 +1,264 @@
+
+            audit_entry(
+                error_message,
+                level='error'
+            )
+
+            raise ValueError(error_message)
+
+        current_migration_id = str(self.get_latest_applied_migration_from_fhir())
+        latest_created_migration_id = str(self.get_latest_created_migration())
+
+        if current_migration_id != latest_created_migration_id:
+            error_message = f"There exists an unapplied migration."
+
+            audit_entry(
+                error_message,
+                level='error'
+            )
+
+            raise RuntimeError(error_message)
+
+        new_id = str(uuid.uuid4())
+        migration_filename = f"{migration_name}.py"
+        migration_path = os.path.join(self.migrations_dir, migration_filename)
+
+        with open(migration_path, "w") as migration_file:
+            migration_file.write(f"# Migration script generated for {migration_name}\n")
+            migration_file.write(f"revision = '{new_id}'\n")
+            migration_file.write(f"down_revision = '{current_migration_id}'\n")
+            migration_file.write("\n")
+            migration_file.write("def upgrade():\n")
+            migration_file.write("    # Add your upgrade migration code here\n")
+            migration_file.write("    print('upgraded')\n")
+            migration_file.write("\n")
+            migration_file.write("def downgrade():\n")
+            migration_file.write("    # Add your downgrade migration code here\n")
+            migration_file.write("    print('downgraded')\n")
+            migration_file.write("\n")
+
+        return migration_filename
+
+    def run_migrations(self, direction: str):
+        """Run migrations based on the specified direction ("upgrade" or "downgrade")."""
+        # Update the migration to acquire most recent updates in the system
+        self.build_migration_sequence()
+        if direction not in ["upgrade", "downgrade"]:
+            raise ValueError("Invalid migration direction. Use 'upgrade' or 'downgrade'.")
+
+        current_migration = self.get_latest_applied_migration_from_fhir()
+        if current_migration and self.migration_sequence.find(current_migration) is None:
+            message = "Applied migration does not exist in the migration system"
+            audit_entry(
+                message,
+                level='info'
+            )
+
+            raise KeyError(message)
+
+        applied_migrations = None
+        unapplied_migrations = None
+        if direction == "upgrade":
+            unapplied_migrations = self.get_unapplied_migrations(current_migration)
+        elif direction == "downgrade" and current_migration is not None:
+            applied_migrations = self.get_previous_migration(current_migration)
+            unapplied_migrations = current_migration
+
+        if not unapplied_migrations or unapplied_migrations == 'None':
+            raise ValueError("No valid migration files to run.")
```

Comment:
why is this an exceptional case?  (to not find any unapplied migrations)  i'd expect a silent run if attempting `upgrade` and finding the system already up to date.

Reply:
Will remove.